update tiny error in voting.md

lyricalpolymath opened the pull request:
changed the exponent
in the current/original wiki it said "This means expressing 50% is `50 * 10 ^ 16`"  
changed to "This means expressing 50% is `50 * 10 ^ 18`"  
to be consistent with the exponent defined a few words before

izqui commented:
That's actually not correct, but maybe it isn't too clear in the documentation.

If 10 ^  18 is 100%, therefore 10 ^ 16 is 1 %, so 50% is 50 * 10 ^ 16.

Do you have an idea how we can rephrase it better so it is clearer? :)

lyricalpolymath commented:
Oh I see... my bad! :)
It's confusing... why would you choose a range between a maximum (10^18) and a minimum (10^16)
rather than simply using 0 as a minimum?
I've had bad rounding situations for ranged percentages.

I've seen in the [Voting.sol code](https://github.com/aragon/aragon-apps/blob/4ed2806606dec45b75aa9885c1ffd7dedcf1be4a/apps/voting/contracts/Voting.sol#L269) the calculation you are doing... **where does this reasoning come from**?
```
uint256 m = _total * _pct;
uint256 v = m / PCT_BASE;
```

2 considerations before answering your question (they go beyond the scope of changing the wiki so if you want we can talk about it elsewhere)

1- **I think it's bad UX** to ask the user to do the mental calculations in scientific notation...
It's completely unintuitive, and a bit unfriendly also because the actual input the user should write is 10000000000000000  for 1%   (unless the web3 interface is converting another simpler number (ie 1) to write the data as the smart contract expects it

It would be better to have the user input an int between 1 and 100 (more natural) 
or between 10 and 1000 if you want 2 decimal precision points (which I think is not necessary) and do all the conversions in the smart contract. 

----------

2- As I've written in this [post in the aragon chat](https://aragon.chat/channel/design?msg=ooLCBdN3jnbnb3Gij) 
I think it would be interesting in the future to **decouple the vote casting from the voting calculation** so that different types of quorums can be plugged into the voting apps as modules

----------

## 3- quick fix to the wiki
if you want to delay this and simply clarify in the wiki I'd give a bit more explanations and give specific examples the user can actually use as inputs... something like this:


### A Note on Percentages
The variables "Support required" and "Minimum acceptance quorum" are percentages that are expressed between a minimum of 10^16 (that represents 1%) and a maximum of 10^18 (that represents 100%)
You should input the actual number, not the the scientific notation
so 10^16   is  10000000000000000      (or 1 with 16 zeros)
    10^18   is  1000000000000000000  (or 1 with 18 zeros)

Here are a few percentages you can use 


Percentage | Scientific Notation | actual input
------------ | ------------- |  ------------- 
1%     | 10^16         | 10000000000000000
10%   | 10*10^16   | 100000000000000000
20%   | 20*10^16   | 200000000000000000
25%   | 25*10^16   | 250000000000000000
50%   | 50*10^16   | 500000000000000000
60%   | 60*10^16   | 600000000000000000
70%   | 70*10^16   | 600000000000000000
75%   | 75*10^16   | 750000000000000000
100% | 100*10^16 | 1000000000000000000

 -----------------
**if you think this las explanation is right I'll correct the Pull Request and resubmit it**

izqui commented:
> It would be better to have the user input an int between 1 and 100 (more natural)
or between 10 and 1000 if you want 2 decimal precision points (which I think is not necessary) and do all the conversions in the smart contract.

That's exactly how it will work under the hood. The reason we use 10^16 as base is to have that many precision points in between two natural number percentages.

> decouple the vote casting from the voting calculation so that different types of quorums can be plugged into the voting apps as modules

This is definitely an interesting idea, let's explore it.

> A Note on Percentages

That will be a great addition to the wiki, since that's a developer reference more than anything :) As I say, users wont see those numbers, ever.

lyricalpolymath commented:
updated.. check if it's right now
I've added a tiny explanation for the user if they happen to read the wiki

> if you are a front end user you can skip this section, but if you are a developer and want to manipulate the [Voting contract](https://github.com/aragon/aragon-apps/blob/master/apps/voting/contracts/Voting.sol) directly these are some notes you should consider.

izqui commented on docs/dev/apps/voting.md:
```diff
@@ -13,11 +13,32 @@ The Voting app is instantiated with a certain set of parameters that won’t be
 
-The only parameter that can be changed if 'Minimum acceptance quorum' for protecting against the case in which there is not enough voter turnout.
+### A Note on Percentages
+if you are a front end user you can skip this section, but if you are a developer and want to manipulate the [Voting contract](https://github.com/aragon/aragon-apps/blob/master/apps/voting/contracts/Voting.sol) directly these are some notes you should consider.
```
Uppercase **I** in 'if', please :)

izqui commented on docs/dev/apps/voting.md:
```diff
@@ -13,11 +13,32 @@ The Voting app is instantiated with a certain set of parameters that won’t be
+if you are a front end user you can skip this section, but if you are a developer and want to manipulate the [Voting contract](https://github.com/aragon/aragon-apps/blob/master/apps/voting/contracts/Voting.sol) directly these are some notes you should consider.
 
-### Vote lifecycle
+The variables "Support required" and "Minimum acceptance quorum" are percentages that are expressed between a minimum of 10^16 (that represents 1%) and a maximum of 10^18 (that represents 100%).
```
Actually 10^16 is not the minimum, the minimum is really 0, which is 0%. Between 0 and 10^16, is 0 to 1% :)

izqui commented:
Thanks so much for the help!
